pyradiosky/skymodel.py

```diff
@@ -1672,53 +1883,39 @@ def assign_to_healpix(
         sky = self if inplace else self.copy()
 
         if frame is None:
-            if sky.frame is None and sky._frame_inst is None:
-                raise ValueError(
-                    "This method requires a coordinate frame but None was supplied "
-                    "and the SkyModel object has no frame attribute set. Please "
-                    "call this function with a specific frame."
-                )
-            elif sky.frame is not None and sky._frame_inst is None:
-                # this is an unexpected state where the frame name is set
-                # but the instance has been destroyed somehow.
-                # Use the name to rebuild the instance
-
-                # easiest way to do frame checking is through making a dummy skycoord
-                coords = SkyCoord(0, 0, unit="deg", frame=self.frame)
-
-                frame = coords.frame
-                sky._frame_inst = frame
-            else:
-                # use the frame associated with the object already
-                frame = sky._frame_inst
+            frame_obj = sky.skycoord.frame
+            sky.hpx_frame = sky.skycoord.frame.name
         else:
+            warnings.warn(
+                "The frame keyword is deprecated, in version 0.3.0 this method will use "
+                "the frame of this object's skycoord attribute.",
+                DeprecationWarning,
+            )
             # easiest way to do frame checking is through making a dummy skycoord
             coords = SkyCoord(0, 0, unit="deg", frame=frame)
 
-            frame = coords.frame
+            frame_obj = coords.frame
 
-            if not isinstance(frame, (Galactic, ICRS)):
+            if not isinstance(frame_obj, (Galactic, ICRS)):
                 raise ValueError(
-                    f"Supplied frame {frame.__class__.__name__} is not supported at "
+                    f"Supplied frame {frame_obj.__class__.__name__} is not supported at "
                     "this time. Only 'galactic' and 'icrs' frames are currently supported.",
                 )
```

Comment:
hmm, I don't think it should be. I'll clean this up.

Reply:
Oh, I see. This input parameter is deprecated, so I didn't add handling for anything new. In the future it will just use the frame from the input object's skycoord parameter.

Reply:
I'll update the docstring to identify that it's deprecated (it already issues a deprecation warning if you use it).

Reply:
sure, I can add that.

Reply:
I realized something as I was thinking about this more. We currently only save the frame name on the healpix object. But if we allow any frame to be converted, we'd need to store more information (like epoch for fk4 or fk5 or an earth location for alt/az). I'm leaning towards going back to only allowing Galactic or ICRS to prevent these worries. I guess I could pretty easily add fk4 or fk5 as long as they had the default epochs for those frames.

Reply:
Plan is to use an Astropy Frame object on healpix objects rather than a string (allow initialization via string). Handle in skyh5 files similarly to the non-healpix handling.

Reply:
done.